programs/ssl/ssl_server2.c

```diff
@@ -3643,6 +3646,11 @@ int main( int argc, char *argv[] )
     }
 #endif
 
+#if defined(MBEDTLS_MEMORY_DEBUG)
+    mbedtls_memory_buffer_alloc_cur_get(&current_heap_memory, &current_heap_blocks);
+    mbedtls_printf( " Heap memory usage after handshake: %zd bytes\n", current_heap_memory);
```

Comment:
Is "after the handshake" here or is it after we verify the client certificate ("5. Verify the client certificate")?

Reply:
I know that verification server certificate is part of handshake, but about verification of the client certificate I am not sure. Anyway I measured also memory usage just before "* 6. Read the HTTP Request" and values are the same as for code above: current memory usage 56024 and for peak memory is 79404 bytes

Reply:
Verifying the client certificate is part of the handshake, but the the comment "5. Verify the client certificate" is misleading. At this point the handshake is already finished and the client certificate is already verified. What happens here is only getting the verification results and printing it out along with the client certificate.

I agree that the safest place to print these things out is just before receiving application data. (That is in the "* 6. Read the HTTP Request" section.)

Reply:
I moved this print before "* 6. Read the HTTP Request" section